github/circleci/README.md

```diff
@@ -0,0 +1,21 @@
+# CircleCI best practices
+
+## Making steps self-explanatory
+
+CircleCI offers a way to make a step in a CI config or org self-explanatory. We SHOULD always use this to make the config more self-explanatory.
```

Comment:
Can we make this short and sweet with more direct language, and by removing a redundant sentence?

Like:

```suggestion
Always set the `name` of your steps to something that explains what it does.
```